types/rdf-dataset-indexed/index.d.ts

```diff
@@ -0,0 +1,10 @@
+// Type definitions for rdf-dataset-indexed 0.4
+// Project: https://github.com/rdf-ext/rdf-dataset-indexed
+// Definitions by: Chris Wilkinson <https://github.com/thewilkybarkid>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { BaseQuad, DataFactory, DatasetCore, Quad } from 'rdf-js';
+
+declare function datasetFactory<Q extends BaseQuad = Quad>(quads?: Q[], dataFactory?: DataFactory): DatasetCore<Q>;
```

Comment:
Would be `DataFactory<Q>` if https://github.com/DefinitelyTyped/DefinitelyTyped/pull/41485 is merged first.